### PR TITLE
Fix gps dot

### PR DESCRIPTION
**Pull request by penev92**
More hacks to fix GPS bogosity.
Should no longer enable auto targeting of actors under shroud.

### CODE REVIEW

File: OpenRA.Mods.RA/Effects/GpsDot.cs [hunk 1]
```diff
@@ -92,17 +100,21 @@ public void Tick(World world)
...
-			show = false;
-			if (!self.IsInWorld || self.IsDead || self.World.RenderPlayer == null)
+			if (!self.IsInWorld || self.IsDead)
 				return;
 
-			var gps = watcher[self.World.RenderPlayer];
-			show = (gps.Granted || gps.GrantedAllies) && ShouldShowIndicator();
+			foreach (var player in self.World.Players)
+			{
+				showToPlayer[player] = false;
```
Comment (pchote) [inline OpenRA.Mods.RA/Effects/GpsDot.cs]: You set this again unconditionally below, so this line is redundant.

**Comment by pchote** [comment]
This fixes a bug that makes gap generators basically useless in the late game, so this should be merged to prep.

**Comment by penev92** [comment]
Updated. I also figured the test commit from #10007 can also help out with the testing of this one.

File: OpenRA.Game/Traits/TraitsInterfaces.cs [hunk 2]
```diff
@@ -189,7 +189,12 @@ public interface IRadarSignature
 	public interface IDefaultVisibilityInfo : ITraitInfo { }
 	public interface IDefaultVisibility { bool IsVisible(Actor self, Player byPlayer); }
 	public interface IVisibilityModifier { bool IsVisible(Actor self, Player byPlayer); }
-	public interface IFogVisibilityModifier { bool HasFogVisibility(Player byPlayer); }
+
+	public interface IFogVisibilityModifier
+	{
+		bool IsVisible(Actor actor);
```
Comment (pchote) [inline OpenRA.Game/Traits/TraitsInterfaces.cs]: This may want to include `Player byPlayer` for consistency.

Comment (pchote) [inline OpenRA.Game/Traits/TraitsInterfaces.cs]: Or maybe we should remove it from `HasFogVisibility` instead.

**Comment by pchote** [comment]
I tested this in a multiplayer game, and AA guns were still able to attack migs that were hidden under a gap generator :(

**Comment by penev92** [comment]
Updated.
FWIW this PR seems to be doing its job well enough. And that is disabling auto targeting under shroud (gap generator) when there is a GPS available.
I believe you are refering to #9999, which is fixed by #10007 and is totally different - it's about firing at an already acquired target that moves into shroud.
Basically both of these PRs each fix half of the problem. For best results you need them both, but for the moment you can test each separately.

**Comment by pchote** [comment]
Code changes look good to me, and confirmed on a branch that also included #10007 that AA guns vs aircraft (finally!) work as expected both with and without GPS. :+1:

**Comment by pchote** [comment]
Adding has dependencies so that this isn't accidentally merged with the testcase commit attached.

File: mods/ra/rules/aircraft.yaml [hunk 3]
```diff
@@ -88,7 +88,7 @@ MIG:
 		Name: Mig Attack Plane
 		Description: Fast Ground-Attack Plane.\n  Strong vs Buildings, Tanks\n  Weak vs Infantry, Light armor, Aircraft
 	Health:
-		HP: 70
+		HP: 700
```
Comment (obrakmann) [inline mods/ra/rules/aircraft.yaml]: This seems rather unrelated.

**Comment by obrakmann** [comment]
Looks good to me, :+1: . Tested in conjunction with #10007 and #9874.

You can remove the [test case](../commit/ff0cdfb49baab882deb69f9d9047628bf6306121) now.

**Comment by penev92** [comment]
Test case removed.

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/53e691963547224977d85559fe5f969770454c88)
